engines/io_uring.c

```diff
@@ -343,6 +365,14 @@ static enum fio_q_status fio_ioring_queue(struct thread_data *td,
 
 	/* ensure sqe stores are ordered with tail update */
 	write_barrier();
+	if (o->cmdprio_percentage && rand_between(&td->prio_state, 0, 99) < o->cmdprio_percentage) {
```

Comment:
I'd put this out of line - which you did in libaio, but make it ala:

```
if (o->cmdprio_percentage)
    fio_ioring_prio_prep(bla);
```

for both of them.